step-framework-parent/step-framework-timeseries/src/main/java/step/core/timeseries/bucket/Bucket.java

```diff
@@ -95,8 +95,11 @@ public void setDistribution(Map<Long, Long> distribution) {
         this.distribution = distribution;
     }
 
-    public long getPercentile(int percentile) {
-        long numberOfPointsAtPercentile = (long) (1.0 * percentile / 100 * count);
+    public long getPercentile(double percentile) {
+        if (percentile < 0.0 || percentile > 100.0) {
+            throw new IllegalArgumentException("Percentile must be between 0.0 and 100.0");
+        }
+        long numberOfPointsAtPercentile = (long) Math.ceil(percentile / 100 * count);
```

Comment:
@jeromecomte until now we were truncating the "index" of the percentile values, since we are doing the distribution by rounding down to the pcl precision. I think it is more correct to round up as done in this PR. What do you think?